pkg/asset/store.go

```diff
@@ -20,23 +20,59 @@ type Store interface {
 	// Fetch retrieves the state of the given asset, generating it and its
 	// dependencies if necessary.
 	Fetch(Asset) error
+
+	// Save dumps the entire state map into a file
+	Save(dir string) error
+
+	// Purge deletes the on-disk assets that are consumed already.
+	// E.g., install-config.yml will be deleted after fetching 'manifests'.
+	Purge(excluded []WritableAsset) error
+}
+
+// assetState includes an asset and a boolean that indicates
+// whether it's dirty or not.
+type assetState struct {
+	asset Asset
+	dirty bool
 }
 
 // StoreImpl is the implementation of Store.
 type StoreImpl struct {
-	assets          map[reflect.Type]Asset
+	assets          map[reflect.Type]assetState
 	stateFileAssets map[string]json.RawMessage
+	fileFetcher     *fileFetcher
```

Comment:
Why is the fileFetcher not storing the files that we consumed by an asset?

Reply:
feel like the whether the asset is loaded or not is determined by the store not the file fetcher, so I'd like to keep it in the store.